File: js/creational/builder.js
class Pizza {
    constructor() {
        this.ingridients = []
    }

    show() {
        console.log(`pizza contains: ${this.ingridients.join(', ')}`)
    }

}

class PizzaBuilder {
    constructor() {
        this.pizza = new Pizza()
    }

    add_cheese() {
        this.pizza.ingridients.push('cheese')
        return this
    }

    add_tomato() {
        this.pizza.ingridients.push('tomato')
        return this
    }

    add_pepperoni() {
        this.pizza.ingridients.push('pepperoni')
        return this
    }

    build() {
        return this.pizza
    }
}

builder = new PizzaBuilder()
const pizza = builder.add_cheese().add_pepperoni().add_tomato().build();
pizza.show()

File: python/creational/builder.py
class Pizza:
    def __init__(self):
        self.ingredients = []

    def show(self):
        print(f'pizza with: {', '.join(self.ingredients)}')


class PizzaBuilder:
    def __init__(self):
        self.pizza = Pizza()

    def add_cheese(self):
        self.pizza.ingredients.append('cheese')
        return self

    def add_tomato(self):
        self.pizza.ingredients.append('tomato')
        return self

    def add_pepperoni(self):
        self.pizza.ingredients.append('pepperoni')
        return self

    def build(self):
        return self.pizza

builder = PizzaBuilder()
pizza = builder.add_cheese().add_tomato().add_cheese().build()

pizza.show()